notes/stv/20191120-openstack-deployment.txt

```diff
@@ -0,0 +1,761 @@
+#-------------------------------------------------------------------------
+
+## for each worker node
+
+sudo yum install python3
+sudo update-alternatives --install /usr/bin/python python /usr/bin/python3 10
+
+
+
+
+#-------------------------------------------------------------------------
+## Start and Stop HDFS services
+#-------------------------------------------------------------------------
+
+
+
+## On Master Node: (ssh stv-aglais-master)
+
+start-all.sh
+
+
+## Create a directory
+hdfs dfs -mkdir /hadoop/
+hdfs dfs -mkdir /hadoop/books/
+
+
+## Run test job
+
+cd /home/hadoop
+wget -O alice.txt https://www.gutenberg.org/files/11/11-0.txt
+wget -O holmes.txt https://www.gutenberg.org/files/1661/1661-0.txt
+wget -O frankenstein.txt https://www.gutenberg.org/files/84/84-0.txt
+
+
+hdfs dfs -put alice.txt holmes.txt frankenstein.txt books
+
+
+
+
+## LS directory
+hdfs dfs -ls /hadoop/books
+
+
+yarn jar ~/hadoop/share/hadoop/mapreduce/hadoop-mapreduce-examples-3.1.3.jar wordcount "/hadoop/books/*" output
+## Job Completed Successfully
+## Check Results
+
+hdfs dfs -ls output
+hdfs dfs -cat output/part-r-00000 | less
+
+
+
+
+
+
+#-------------------------------------------------------------------------
+# Spark
+#-------------------------------------------------------------------------
+
+## Setup Spark on Existing yarn & hdfs cluster
+
+
+## Download and Install Spark Binaries
+## On Master Node: (ssh stv-aglais-master)
+
+cd /home/fedora
+wget https://www-eu.apache.org/dist/spark/spark-2.4.4/spark-2.4.4-bin-hadoop2.7.tgz
+tar -xvf spark-2.4.4-bin-hadoop2.7.tgz
+mv spark-2.4.4-bin-hadoop2.7.tgz spark
+
+
+## Setup Default Spark Config
+
+mv /home/fedora/spark/conf/spark-defaults.conf.template /home/fedora/spark/conf/spark-defaults.conf
+
+cat <<EOF >> "/home/fedora/spark/conf/spark-defaults.conf"
+spark.master                     yarn
+spark.driver.memory              50g
+spark.yarn.am.memory            50g
+spark.executor.memory          50g
+spark.eventLog.enabled  true
+spark.eventLog.dir hdfs://${stv-aglais-master-ip:?}:9000/spark-log
+EOF
+
+
+## Create the log directory in HDFS:
+hdfs dfs -mkdir /spark-log
+
+
+./bin/spark-submit --class org.apache.spark.examples.SparkPi     --master yarn-client     --num-executors 1     --driver-memory 512m     --executor-memory 512m     --executor-cores 1     examples/jars/spark-examples*.jar 10
+
+   Exception in thread "main" java.lang.IllegalArgumentException: Required AM memory (46080+4608 MB) is above the max threshold (50000 MB) of this cluster! Please check the values of 'yarn.scheduler.maximum-allocation-mb' and/or 'yarn.nodemanager.resource.memory-mb'.
+
+
+## Change Spark Configuration to 
+
+spark.master                     yarn
+spark.driver.memory              40g
+spark.yarn.am.memory            40g
+spark.executor.memory          40g
+spark.eventLog.enabled  true
+spark.eventLog.dir hdfs://${stv-aglais-master-ip:?}:9000/spark-log
```

Comment:
Does the Spark config accept and resolve the `${stv-aglais-master-ip:?}`, or is this a place holder for a manual edit ?